lib/archive/util.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2016-2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package archive
+
+import (
+	"path/filepath"
+	"strings"
+)
+
+const (
```

Comment:
This and the GenerateFilterSpec function seem unnecessarily complex. I think the consumer should just call GenerateCopy<direction>FilterSpec functions since that indicates direction.